### PR TITLE
4.) refactoring allen smFish with new spot finding

### PR DESCRIPTION
**Pull request by shanaxel42**
This PR:
- adds a version of TrackpyLocalMaxPeakFinder to the new FindSpots module that process SpotFindingResults
- Adds a new trace building strategy to use in decoding called SEQUENTIAL that builds spot traces  without merging across channels and imaging rounds. Used for sequential methods like smFIsh.
- refactors smFish to use FindSpots.TrackpyLocalMaxPeakFinder DecodeSpots.PerRoundMaxChannel with the trace building strategy SEQUENTIAL

depends on: #1592 #1518 #1517

### CODE REVIEW

File: starfish/core/spots/FindSpots/trackpy_local_max_peak_finder.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    This is a wrapper for :code:`trackpy.locate`, which implements a version of the
+    `Crocker-Grier <crocker_grier>`_ algorithm.
+
+    .. _crocker_grier: https://physics.nyu.edu/grierlab/methods3c/
+
+    Parameters
+    ----------
+
+    spot_diameter : odd integer or tuple of odd integers.
```
Comment (ttung) [inline starfish/core/spots/FindSpots/trackpy_local_max_peak_finder.py]: 1. params are not actually typed to match what is in the description.
2. the params' types in the documentation does not follow the standard style we use (which is whatever is the typing in the actual function declaration).
3. the styling for defaults does not match the styling for defaults in earlier PRs.
4. some defaults are missing.

File: starfish/test/full_pipelines/api/test_allen_smFISH.py [hunk 2]
```diff
@@ -4,10 +4,11 @@
 import starfish.data
 from starfish import FieldOfView
 from starfish.image import Filter
-from starfish.spots import DetectSpots
+from starfish.spots import FindSpots
+from starfish.types import TraceBuildingStrategies
 
 
-@pytest.mark.skip('issues with checksums prevent this data from working properly')
+@pytest.mark.skip('This test runs but takes forever')
```
Comment (ttung) [inline starfish/test/full_pipelines/api/test_allen_smFISH.py]: hahahahaha

For what it's worth, there are a few allen smfish FOVs that have relatively few spots and run relatively quickly (2-3 minutes?)

File: notebooks/py/smFISH.py [hunk 3]
```diff
@@ -123,6 +125,11 @@ def processing_pipeline(
     print("Loading images...")
     images = enumerate(experiment[fov_name].get_images(FieldOfView.PRIMARY_IMAGES))
 
+    decoder = starfish.spots.DecodeSpots.PerRoundMaxChannel(
```
[the remark this answers is not in the document]
Comment (shanaxel42) [reply notebooks/py/smFISH.py]: I wanted to put it above but there's no access to the experiment.codebook at that point

File: starfish/test/full_pipelines/api/test_allen_smFISH.py (hunk 2 above)
Comment (shanaxel42) [reply starfish/test/full_pipelines/api/test_allen_smFISH.py]: Do you know which ones?

File: notebooks/py/smFISH.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ttung) [reply notebooks/py/smFISH.py]: Ahh, I see.

File: starfish/test/full_pipelines/api/test_allen_smFISH.py (hunk 2 above)
Comment (ttung) [reply starfish/test/full_pipelines/api/test_allen_smFISH.py]: fov_004 has very few spots.

**Pull request merged**